assets/js/googlesitekit/datastore/user/feature-tours.js

```diff
@@ -80,19 +90,69 @@ const baseActions = {
 		invariant( slug, 'A tour slug is required to dismiss a tour.' );
 
 		return ( function* () {
+			const { select } = yield getRegistry();
+			if ( select( STORE_NAME ).isFetchingDismissTour( slug ) ) {
+				const response = select( STORE_NAME ).getDismissedTours();
+				return { response, error: undefined };
+			}
 			// Dismiss the given tour immediately.
 			yield {
 				payload: { slug },
 				type: DISMISS_TOUR,
 			};
-			// Dispatch a request to persist the dismissal.
-			const { response, error } = yield fetchDismissTour( slug );
-			if ( ! error ) {
-				yield receiveGetDismissedTours( response );
-			}
-			return { response, error };
+			// Dispatch a request to persist and receive updated dismissed tours.
+			return yield fetchDismissTour( slug );
 		}() );
 	},
+
+	receiveReadyTours( readyTours, { viewContext } = {} ) {
+		invariant( Array.isArray( readyTours ), 'readyTours must be an array.' );
+		invariant( viewContext, 'viewContext is required.' );
+		return {
+			payload: { readyTours, viewContext },
+			type: RECEIVE_READY_TOURS,
+		};
+	},
+
+	receiveTours( tours ) {
+		invariant( Array.isArray( tours ), 'tours must be an array.' );
+		return {
+			payload: { tours },
+			type: RECEIVE_TOURS,
+		};
+	},
```

Comment:
This appears to be unnecessary, and it isn't used since we can import the available feature tours directly. Let's remove it.

Reply:
It's used in tests right now which is necessary for being able to test with different tours.